### PR TITLE
mgmt aks, support beginCreateAgentPool

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

fix https://github.com/Azure/azure-sdk-for-java/issues/30374

Based on draft PR https://github.com/Azure/azure-sdk-for-java/pull/30073, added comments and update session-records.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    public AgentPoolData() {
+        this(new AgentPoolInner());
+    }
+
+    /**
+     * Creates an instance of agent pool data.
+     *
+     * @param innerModel the inner model of agent pool.
+     */
+    public AgentPoolData(AgentPoolInner innerModel) {
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java:37]: Not sure if we want to expose this constructor to user. Tried some alternatives but failed checkstyle.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java:37]: You may try to have another `AgentPoolDataImpl` in implementation package (inherit this one), and in KubernetesClusterImpl.java, init the `AgentPoolDataImpl` with this additional ctor.

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * The client-side data of an agent pool.
+ */
+public class AgentPoolData implements AgentPool, HasInnerModel<AgentPoolInner> {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java:21]: Any suggestion for the name? AgentPoolRequest? AgentPoolContent?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java:21]: AgentPoolData sounds ok to me. But if we want another class in `implementation` package to return to the user, we may call this `AgentPoolRequest` and that in `implementation` `AgentPoolData`?

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java:21]: Anything in `implementation` package should not affect naming of things in exposed models (implementation is not supposed to be exposed, and you can even name the subclass same `AgentPoolData` or `MonkeyData` in implementation and still go on with it)

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/AgentPoolDataImpl.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.azure.resourcemanager.containerservice.implementation;
+
+import com.azure.resourcemanager.containerservice.fluent.models.AgentPoolInner;
+import com.azure.resourcemanager.containerservice.models.AgentPoolData;
+
+/**
+ * AgentPool implementation that exposes AgentPoolInner as constructor arguments.
+ */
+public class AgentPoolDataImpl extends AgentPoolData {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/AgentPoolDataImpl.java:9]: package private?

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/AgentPoolDataImpl.java (hunk 3 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/AgentPoolDataImpl.java]: Changed.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**